pyrtl/helperfuncs.py

```diff
@@ -795,11 +795,20 @@ def _convert_verilog_str(val: str, bitwidth: int = None,
     if val.startswith('-'):
         neg = True
         val = val[1:]
+
     split_string = val.lower().split("'")
     if len(split_string) != 2:
         raise PyrtlError('error, string not in verilog style format')
     try:
         bitwidth = int(split_string[0])
+        if passed_bitwidth is not None:
+            if bitwidth > passed_bitwidth:
+                raise PyrtlError(
+                    "bitwidth parameter passed (%d) cannot fit Verilog-style constant with bitwidth %d"
+                        % (passed_bitwidth, bitwidth)
+                )
+            bitwidth = passed_bitwidth
```

Comment:
This is the inverse of line 792 above :) I think we can delete lines 792 and 810 if we change line 803 to `verilog_bitwidth = int(split_string[0])`, since the Verilog bitwidth doesn't seem to be used outside of this code block. 

I think this also lets you combine the two `if` statements on lines 804 and 805 into `if bitwidth is not None and verilog_bitwidth > bitwidth:`